uncch-rdmc/dataverse-frontend
# Added locales (lng files) and language switcher

**What this PR does / why we need it:**

This pull request introduces French (fr) locale support and implements a language switcher component. These changes are necessary to provide a localized experience for French-speaking users and to allow users to easily switch between available languages.

**Special notes for your reviewer:**

- Please review the accuracy of the French translations in the added JSON files.
- Verify that the language switcher component functions correctly and seamlessly switches between languages.
- Ensure that the addition of the French locale does not introduce any bugs or issues in the existing codebase.

**Suggestions on how to test this:**

- Checkout the branch associated with this pull request.
- cd dev-env (ensure the dvconfig.py contains the dataverse API key)
- Navigate to the header and locate the language switcher component (dropdown UI based on design system)
- Test switching between the available languages (including French) and verify that the content updates accordingly.
- Review the French translations in various parts of the application to ensure accuracy and completeness; available at public/locales/fr

**Does this PR introduce a user interface change?**

Yes, this PR introduces a language switcher component in the header. 

**Is there a release notes update needed for this change?:**

Yes, the release notes should mention the addition of French locale support and the introduction of the language switcher feature.

**Additional documentation:**

- Update the project's README file to mention the supported languages and how to contribute translations for new languages.
- Add documentation on how to provide feedback for the language translation from localized users.
